jquery.xblock.js

```diff
@@ -299,10 +299,29 @@
             return cookieOptions;
         },
 
+        get_querystring: function(){
+            // For easier test stubbing.
+            return location.search;
+        },
+
         init: function(options, root) {
             var $this = this,
                 deferred = $.Deferred(),
-                blockURL = this.getViewUrl('student_view', options);
+                blockURL = this.getViewUrl('student_view', options),
+                data = options.data;
+
+            // http://stackoverflow.com/a/901144/882918
+            function getParameterByName(name) {
+                name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+                var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
+                    results = regex.exec($this.get_querystring());
+                return results === null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
+            }
+
+            var activateChild = getParameterByName('activate');
+            if (activateChild) {
+                data['activate_block_id'] = activateChild;
```

Comment:
It would be clearer to update `options.data` explicitly, to know what you are doing here exactly without having to refer to the declaration of `data`.